docs/design/proposal-troubleshoot-consolidated-cli.md

```diff
@@ -0,0 +1,199 @@
+# Consolidated Troubleshoot CLI
+
+## Goals
+
+As Troubleshoot grows and gains new features, some of which involves flags for setting options, and to make it easier to add additional subcommands that don't belong underneath either `support-bundle` or `preflight` binaries, we would like to consolidate all of the Troubleshoot commands under one binary/plugin.
+
+There is discussion about changing the behaviour of `preflight`, considering that preflights and support-bundles utilize the same specs - the same collectors and analyzers - and only differ in what is returned to the user post-analysis.  For this design proposal, `support-bundle` and `preflight` may be condensed into a single `troubleshoot` binary/plugin.
+
+## Non-Goals
+
+## Background
+
+## High Level Design
+
+Functions of `support-bundle`, `preflight`, `analyze`, `redact`, and `sbctl` binaries/tools should be rolled together into a single `troubleshoot` CLI plugin that can perform all necessary functions of the same.
+
+`troubleshoot` CLI plugin should be able to report on the version that is installed in the CLI, and any support bundle generated with `troubleshoot` should report its build/version inside the archive it generates.
+
+## Detailed Design
+
+- TODO: specify the public APIs we expect users of Troubleshoot to consume, e.g.
+  - collect()
+  - redact()
+  - analyze()
+
+In the interest of being able to work on this quickly without breaking existing use-cases, a new `troubleshoot` command should be created. Utilizing cobra and viper best practices from the cobra.dev docs.
+
+sbctl should be migrated to the troubleshoot repository in a "lift and shift" operation to start with, allowing for it's methods and functions to be called by the new `troubleshoot` command
+
+### Public APIs
+
+The existing package entrypoint we use from the support-bundle command is supportbundle.CollectSupportBundleFromSpec()
+which is an end-to-end function that runs collectors, redactors and then analysis.
+
+The Consolidated cli should be responsible for handling the end-to-end running of this process in the `cmd` leaving the importable `pkg` free to focus on defining the individual collect, redact analyze steps.
+
+this allows us to move functions out from behind `CollectSupportBundleFromSpec` so they can be easier called from other projects importing troubleshoot.
+
+those functions could be then called as:
+
+- `troubleshoot.Collect()`
+- `troubleshoot.Redact()`
+- `troubleshoot.Analyze()`
+
```

Comment:
suggest in order to deprecate the `CollectSupportBundleFromSpec()` function, we replace it with a wrapper that calls the cmd - so that to someone importing the function it looks the same, but the logic inside is moved to a more suitable place

Reply:
if we intend to have an end-to-end process function importable like this we could just leave this alone? 

my thoughts were that the end-to-end process should live in the `cmd` instead which isn't the importable bit.